### PR TITLE
WIP: Multisampling for framebuffers

### PR DESCRIPTION
**Pull request by Botje**
I wrote code to enable multisampling for fullscreen framebuffers.
Is there interest in merging this?
I only tested with the shader renderer for now.
This is an OpenGL-only feature, to restore anti-aliasing for people who force it on in their drivers.

Before: https://www.dropbox.com/s/6933731ffq4scu7/without_multisampling.png?dl=0
After: https://www.dropbox.com/s/3iwsxt8kw9gpg5a/with_multisampling.png?dl=0

Known issues:
- [x] Probably linking issues on Windows
- [x] No fallback for Amiga
- [ ] No fallback or error checking in general.
- [x] ADDED: `GfxOpenGLShaders::storeDisplay` is broken (cannot `glReadPixels` from a MS framebuffer

### CODE REVIEW

File: graphics/opengles2/multisampleframebuffer.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	glDeleteRenderbuffers(1, &_msColorId);
+	glDeleteRenderbuffers(1, &_msDepthId);
+	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+	glDeleteFramebuffers(1, &_msFrameBufferId);
+}
+
+//TODO: fallback to fewer samples
+void MultiSampleFrameBuffer::init() {
+	glGenFramebuffers(1, &_msFrameBufferId);
+  glBindFramebuffer(GL_FRAMEBUFFER, _msFrameBufferId);
```
Comment (klusark) [inline graphics/opengles2/multisampleframebuffer.cpp]: Indentation is a bit off.

**Comment by bgK** [comment]
Sounds like a good idea to have this as a replacement to the "antialiasing" configuration key when rendering to framebuffers.

**Comment by Botje** [comment]
So my frustration with GL extensions had the side effect of using glew first, when possible.
If we add GLEW as a build dependency for OpenGL we can get rid of all that nasty pointer grabbing code , but that _does_ require a glew port for AmigaOS or some more `#ifdef`ing.

**Comment by bgK** [comment]
Requiering GLEW as a dependency for "Advanced" OpenGL support sounds sensible to me. Anyway AmigaOS probably does not support any of these exensions, and would not need the extra functions.

**Comment by Botje** [comment]
There are quite a few problems with this approach, sadly.
One big issue with the multisample framebuffer is that you cannot immediately `glReadPixels` from it without transferring the contents to a regular texture-backed framebuffer.
Quite a few functions take image data from the framebuffer and re-render it later, which means unbind-copy-upload-rebind in several places, adding quite a bit of coupling to the renderer and any framebuffer present.
One way out is using framebuffer data from a previous frame, but that will cause artifacts in some places.
Maybe we should just render in an increased resolution instead, that seems like a far easier approach.

Any thoughts on mainlining the GLEW commits? It doesn't have to be a dependency, but making use of it when present does make for easier maintenance.

**Comment by aquadran** [comment]
GLEW commits looks ok

**Comment by Botje** [comment]
I drastically re-did the implementation, now it reads from the framebuffer used in the last frame.
I checked the uses of `glReadPixels` in our code, and it is only used when taking screenshots and dimming the screen, both of which are not required to be frame perfect IMO.
Now all that is left is a fallback in case the extensions needed are not possible (a small refactor, I think), and this should be good to add.

Any opinions on making antialiasing default to true with this?

**Comment by bgK** [comment]
Having FSAA enabled has no visible effect with Myst III, but shouldn't hurt performance too much either. I don't have strong opinions about having anti-aliasing enabled by default.

**Comment by bgK** [comment]
Superseeded by #1250.

**Pull request closed without merging**